src/golang/lib/engine/aq_engine.go

```diff
@@ -186,6 +186,37 @@ func (eng *aqEngine) ExecuteWorkflow(
 		return shared.FailedExecutionStatus, errors.Wrap(err, "Error reading latest workflowDag.")
```

Comment:
just to confirm, this method will be called in `main.go` or `executor` binary, and if something's wrong when spinning up the executor, we will still get an untrackable failure? And I believe we would need go-routine based scheduling to solve this, which is def. out-of-scope for this PR.

Reply:
Right, this decreases the amount of code that would result in an untrackable failure, but we could definitely do more as you said.